src/openassetio-python/package/openassetio/test/manager/apiComplianceSuite.py

```diff
@@ -708,6 +712,493 @@ def __testGetWithRelationshipsError(
+
+        [pager] = self.__test_getWithRelationshipPaged_success([a_ref], a_rel)
+
+        actual = self._concat_all_pages(pager)
+        self.assertListEqual(actual, expected)
+
+    def test_when_relationship_trait_set_known_and_props_set_then_filtered_refs_returned(self):
+        a_ref = self.requireEntityReferenceFixture("a_reference", skipTestIfMissing=True)
+        a_rel = self.requireFixture("a_relationship_traits_data_with_props")
+        expected = self.requireEntityReferencesFixture("expected_related_entity_references")
+
+        [pager] = self.__test_getWithRelationshipPaged_success([a_ref], a_rel)
+
+        actual = self._concat_all_pages(pager)
+        self.assertListEqual(actual, expected)
+
+    def test_when_result_trait_set_supplied_then_filtered_refs_returned(self):
+        a_ref = self.requireEntityReferenceFixture("a_reference", skipTestIfMissing=True)
+        a_rel = TraitsData(self.requireFixture("a_relationship_trait_set"))
+        result_trait_set = self.requireFixture("an_entity_trait_set_to_filter_by")
+        expected = self.requireEntityReferencesFixture("expected_related_entity_references")
+
+        [pager] = self.__test_getWithRelationshipPaged_success(
+            [a_ref], a_rel, resultTraitSet=result_trait_set
+        )
+
+        actual = self._concat_all_pages(pager)
+        self.assertListEqual(actual, expected)
+
+    def test_when_querying_missing_reference_then_resolution_error_is_returned(self):
+        self.__test_getWithRelationshipPaged_error(
+            entity_reference=self.requireEntityReferenceFixture(
+                "a_reference_to_a_missing_entity", skipTestIfMissing=True
+            ),
+            relationship_trait_set=self.requireFixture("a_relationship_trait_set"),
+            expected_error_code=BatchElementError.ErrorCode.kEntityResolutionError,
+            expected_error_message=self.requireFixture("expected_error_message"),
+        )
+
+    def test_when_querying_malformed_reference_then_malformed_reference_error_is_returned(self):
+        self.__test_getWithRelationshipPaged_error(
+            entity_reference=self.requireEntityReferenceFixture(
+                "a_malformed_reference", skipTestIfMissing=True
+            ),
+            relationship_trait_set=self.requireFixture("a_relationship_trait_set"),
+            expected_error_code=BatchElementError.ErrorCode.kMalformedEntityReference,
+            expected_error_message=self.requireFixture("expected_error_message"),
+        )
+
+    def test_when_related_entities_span_multiple_pages_then_pager_has_multiple_pages(self):
+        a_ref = self.requireEntityReferenceFixture("a_reference", skipTestIfMissing=True)
+        a_rel = TraitsData(self.requireFixture("a_relationship_trait_set"))
+        expected_related_refs = self.requireEntityReferencesFixture(
+            "expected_related_entity_references"
+        )
```

Comment:
[question] Any suggestions for better fixture name(s) to make it clear that at least two related refs are required?  Or is the test name and/or assertion check below sufficient enough of a clue?

Reply:
maybe just `a_related_relationship_trait_set` ?

Reply:
Is "related relationship" a bit of a tautology?  I wonder if this comment is less of an issue now that the tests of the 4 methods have been combined?